src/appsignal/agent.py

```diff
@@ -6,9 +6,31 @@
 from .config import Config
 
 AGENT_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "appsignal-agent")
+agent_active = False
+
+
+def is_agent_active():
+    global agent_active
+    return agent_active
+
+
+def agent_is_active():
+    global agent_active
+    agent_active = True
 
 
 def start_agent(config: Config):
     config.set_private_environ()
-
-    subprocess.Popen([AGENT_PATH, "start", "--private"])
+    p = subprocess.Popen(
+        [AGENT_PATH, "start", "--private"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
```

Comment:
This potential issue was highlighted by @luismiramirez on [this Slack message](https://appsignal.slack.com/archives/C055A2H9NPJ/p1684328919356849?thread_ts=1684315640.997929&cid=C055A2H9NPJ):

> This will deadlock when using stdout=PIPE or stderr=PIPE and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use Popen.communicate() when using pipes to avoid that.

The quote is in the context of using `Popen.communicate()` instead of `Popen.wait()`, to avoid a deadlock between the subprocess (waiting for the pipe to be read from so it can write more to it) and the Python application (waiting for the subprocess to end)

But what concerns me is not that we wait for it on the Python side, but that regardless, because we're not reading from that pipe, the pipe becomes full -- causing the agent subprocess to become locked trying to write from it. Is that something that could happen?

Reply:
The agent will not run for a long time. It either exits and prints a couple lines of output, or it starts, daemonizes and starts a new process. I don't think the deadlock because of high output will ever happen.

Reply:
> or it starts, daemonizes and starts a new process

And, just to be sure, the new process wouldn't inherit this stdout/stderr file descriptor, right?

Reply:
I don't think so. But it also doesn't print anything after starting.

Reply:
Ah, yeah, because it always writes to a file, right? That's probably okay then.